### PR TITLE
updated to Swift 4.2 and added swift version to podspec

### PR DESCRIPTION
**Pull request by alexanderkhitev**

### CODE REVIEW

**Comment by alexanderkhitev** [comment]
Hello @mtnbarreto ! Could you check this pull request?

**Comment by 70m3n** [comment]
I would also like to see this merge come through.

**Comment by Climbatize** [comment]
I decided to temporarily use Alexander repo directly, you can do it like this if you want :

pod 'XLPagerTabStrip', git: 'https://github.com/alexanderkhitev/XLPagerTabStrip', branch: 'master'

File: Sources/ButtonBarViewCell.swift [hunk 1]
```diff
@@ -33,8 +33,7 @@ open class ButtonBarViewCell: UICollectionViewCell {
         super.init(coder: aDecoder)
         
         isAccessibilityElement = true
-        accessibilityTraits |= UIAccessibilityTraitButton
-        accessibilityTraits |= UIAccessibilityTraitHeader
+        accessibilityTraits = [.button, .header]
```
Comment (mats-claassen) [inline Sources/ButtonBarViewCell.swift]: I think you should use the `insert` or `formUnion` function here. The previous logic was adding the two traits to whatever was already in accessibilityTraits.

File: Sources/ButtonBarViewCell.swift [hunk 2]
```diff
@@ -44,9 +43,9 @@ open class ButtonBarViewCell: UICollectionViewCell {
         set {
             super.isSelected = newValue
             if (newValue) {
-                accessibilityTraits |= UIAccessibilityTraitSelected
+                accessibilityTraits = [.selected]
```
Comment (mats-claassen) [inline Sources/ButtonBarViewCell.swift]: Again. use `accessibilityTraits.insert(.selected)`

File: Sources/ButtonBarViewCell.swift [hunk 3]
```diff
@@ -44,9 +43,9 @@ open class ButtonBarViewCell: UICollectionViewCell {
...
             } else {
-                accessibilityTraits &= ~UIAccessibilityTraitSelected
+                accessibilityTraits.formUnion(.selected)
```
Comment (mats-claassen) [inline Sources/ButtonBarViewCell.swift]: And here you should remove selected and not add it: `accessibilityTraits.remove(.selected)`

File: Sources/ButtonBarPagerTabStripViewController.swift [hunk 4]
```diff
@@ -349,8 +349,7 @@ open class ButtonBarPagerTabStripViewController: PagerTabStripViewController, Pa
         }
         cell.isAccessibilityElement = true
         cell.accessibilityLabel = cell.label.text
-        cell.accessibilityTraits |= UIAccessibilityTraitButton
-        cell.accessibilityTraits |= UIAccessibilityTraitHeader
+        cell.accessibilityTraits = [.button, .header]
```
Comment (mats-claassen) [inline Sources/ButtonBarPagerTabStripViewController.swift]: Same comment as below

Comment (alexanderkhitev) [reply Sources/ButtonBarPagerTabStripViewController.swift]: @mats-claassen thanks, it done

**Pull request merged**